torch/csrc/inductor/aoti_runtime/model.h

```diff
@@ -303,6 +309,9 @@ class AOTInductorModel : public AOTInductorModelBase<AOTInductorModel> {
       std::optional<std::string> cubin_dir) {
     return std::make_unique<AOTInductorModel>(constants, cubin_dir);
   }
+
+ private:
+  std::unique_ptr<AOTInductorModelKernelsBase> kernels_;
```

Comment:
This is fine, but I wonder if making this a straight member (not a pointer) also works (with `AOTInductorModelKernels` as a forward declaration)?

Reply:
Yup, I had this in a previous iteration. Don't have a strong opinion, let me change back to forward declaration.

Reply:
Seems like even with forward declaration I used a pointer here. It actually makes sense, because C++ allows only pointers and references for incomplete types.